### ISSUE TITLE
Improve progress bar for concurrent downloads

### ISSUE THREAD

**Issue by kalefranz** [conda/conda#5198]
Previous work in https://github.com/conda/conda/pull/3112, but an ncurses requirement on Windows probably won't work.

Try `tqdm`.

* http://danshiebler.com/2016-09-14-parallel-progress-bar/
* https://renzo.lucioni.xyz/parallel-progress/
* https://github.com/tqdm/tqdm/issues/97

**Comment by jakirkham** [comment]
What sort of progress bar is being used currently?

**Comment by kalefranz** [comment]
I *really* would like conda to get to a place where we can have more concurrency in everything.  Both during the download-extract phase, and during the unlink-link phase.  What I envisioned when I created this ticket was a docker-esque progressbar.  I googled around a bit looking for the library docker uses for this.  It might be https://github.com/gosuri/uiprogress.  At least it's a good example.

I guess that turns out to be incredibly hard with python.  Because I haven't seen anything like it built yet.  Not even tqdm (which is what we're now using as of conda 4.4).
